clinica/iotools/converters/adni_to_bids/adni_modalities/adni_fmap.py

```diff
@@ -0,0 +1,338 @@
+                                json_filepath = fmap_filename[:-14] + "fmap_ph.json"
+                                with open(json_filepath, 'r') as file:
+                                    fmap_data = json.load(file)
+                                    if "EchoTime1" in fmap_data and "EchoTime2" in fmap_data:
+                                        os.rename(fmap_filename, fmap_filename[:-14] + "phasediff.nii.gz")
+                                        os.rename(json_filepath, json_filepath[:-12] + "phasediff.json")
+                                    else:
+                                        os.rename(fmap_filename, fmap_filename[:-14] + "phase1.nii.gz")
+                                        os.rename(json_filepath, json_filepath[:-12] + "phase1.json")
+                            elif os.path.exists(fmap_filename[:-17] + "phasediff.json"):
+                                json_filepath = fmap_filename[:-17] + "phasediff.json"
+                                with open(json_filepath, 'r') as file:
+                                    fmap_data = json.load(file)
+                                    if "EchoTime1" in fmap_data and "EchoTime2" in fmap_data:
+                                        os.rename(fmap_filename, fmap_filename[:-17] + "phasediff.nii.gz")
+                        if filename[-17:] == "fmap_e2_ph.nii.gz":
+                            if os.path.exists(fmap_filename[:-17] + "phase2.json"):
+                                os.rename(fmap_filename, fmap_filename[:-17] + "phase2.nii.gz")
+                            elif os.path.exists(fmap_filename[:-17] + "fmap_e2_ph.json"):
+                                json_filepath = fmap_filename[:-17] + "fmap_e2_ph.json"
+                                with open(json_filepath, 'r') as file:
+                                    fmap_data = json.load(file)
+                                    if "EchoTime1" in fmap_data and "EchoTime2" in fmap_data:
+                                        os.rename(fmap_filename, fmap_filename[:-17] + "phasediff.nii.gz")
+                                        os.rename(json_filepath, json_filepath[:-15] + "phasediff.json")
+                                    else:
+                                        os.rename(fmap_filename, fmap_filename[:-17] + "phase2.nii.gz")
+                                        os.rename(json_filepath, json_filepath[:-15] + "phase2.json")
+                            elif os.path.exists(fmap_filename[:-17] + "phasediff.json"):
+                                json_filepath = fmap_filename[:-17] + "phasediff.json"
+                                with open(json_filepath, 'r') as file:
+                                    fmap_data = json.load(file)
+                                    if "EchoTime1" in fmap_data and "EchoTime2" in fmap_data:
+                                        os.rename(fmap_filename, fmap_filename[:-17] + "phasediff.nii.gz")
+
+def check_json(fmap_filename, filename, subroot):
```

Comment:
```suggestion
def check_json(fmap_filename: Path, filename: Path, subroot: Path):
```

Reply:
Noted, and will also comment that a separate function could consolidate the for loop here, or as you mentioned previously perhaps there is a better way